clients/cli/src/copy.rs

```diff
@@ -23,76 +25,243 @@ pub fn copy(path: PathBuf, import_dest: &str, edit: bool) {
-            Ok(file_metadata) => file_metadata,
-            Err(err) => match err {
-                CoreError::UiError(CreateFileAtPathError::FileAlreadyExists) => {
-                    if edit {
-                        get_file_by_path(&get_config(), &import_dest_with_filename).unwrap_or_else(|get_err| match get_err {
-                            CoreError::UiError(GetFileByPathError::NoFileAtThatPath) |
-                            CoreError::Unexpected(_) => exit_with(
-                                &format!("Unexpected error: {:?}", get_err),
-                                UNEXPECTED_ERROR,
+        if !children_paths.is_empty() {
+            for child in children_paths {
+                let child_path = child.path();
+                let child_name = child_path
+                    .file_name()
+                    .and_then(|child_name| child_name.to_str())
+                    .unwrap_or_else(|| {
+                        exit_with(
+                            &format!(
+                                "Failed to read child name, OS parent path: {:?}",
+                                child_path
                             ),
-                        })
-                    } else {
-                        exit_with(&format!("Input destination {} not available within lockbook, use --edit to overwrite the contents of this file!", import_dest_with_filename), FILE_ALREADY_EXISTS)
+                            COULD_NOT_READ_OS_CHILDREN,
+                        )
+                    });
+
+                let ends_with_slash = import_dest.ends_with('/');
+                let lockbook_child_path = if is_top_folder {
+                    let parent_name = path
+                        .file_name()
+                        .and_then(|parent_name| parent_name.to_str())
+                        .unwrap_or_else(|| {
+                            exit_with(
+                                &format!("Failed to read parent name, OS path: {:?}", path),
+                                COULD_NOT_READ_OS_CHILDREN,
+                            )
+                        });
+
+                    format!(
+                        "{}{}{}/{}",
+                        import_dest,
+                        if ends_with_slash { "" } else { "/" },
+                        parent_name,
+                        child_name
+                    )
+                } else {
+                    format!(
+                        "{}{}{}",
+                        import_dest,
+                        if ends_with_slash { "" } else { "/" },
+                        child_name
+                    )
+                };
+
+                recursive_copy_folder(&child_path, &lockbook_child_path, config, edit, false);
+            }
+        } else if let Err(err) = create_file_at_path(config, &import_dest) {
+            match err {
+                CoreError::UiError(CreateFileAtPathError::FileAlreadyExists) => {
```

Comment:
I think all of these match cases that specify `CoreError::UiError` throughout would be more readable as nested under a `UiError` match (as follows), but it's up to you.
```
match err {
    CoreError::UiError(err) => match err {
        CreateFileAtPathError::Whatever => ...,
        ...
    }
    CoreError::Unexpected(msg) => ...,
}
```